### DISCUSSION TITLE
optimization: class duplicates

### DISCUSSION THREAD

**Discussion by SailorMax** [fomantic/Fomantic-UI#1630] [General] [answered]
# Bug Report
Some elements defined with many duplicate class names.
For example: `.ui.primary.loading.loading.loading.loading > i.icon:after`
Do you really need so many ".loading" in same tag?

## Steps to reproduce
loade.css:294

## Expected result
.ui.primary.loading > i.icon:after

## Actual result
.ui.primary.loading.loading.loading.loading > i.icon:after

## Version
2.8.6-dev

**Comment by ColinFrick** [comment, marked as answer]
Yes, this is needed. With an increased specificity we were able to remove most `!important` uses.

For more information see https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity
